telegram/_files/file.py

```diff
@@ -197,14 +187,66 @@ async def download(
         filename.write_bytes(buf)
         return filename
 
-    def _get_encoded_url(self) -> str:
-        """Convert any UTF-8 char in :obj:`File.file_path` into a url encoded ASCII string."""
-        sres = urllib_parse.urlsplit(str(self.file_path))
-        return urllib_parse.urlunsplit(
-            urllib_parse.SplitResult(
-                sres.scheme, sres.netloc, urllib_parse.quote(sres.path), sres.query, sres.fragment
+    async def download_to_object(
+        self,
+        out: IO,
+        read_timeout: ODVInput[float] = DEFAULT_NONE,
+        write_timeout: ODVInput[float] = DEFAULT_NONE,
+        connect_timeout: ODVInput[float] = DEFAULT_NONE,
+        pool_timeout: ODVInput[float] = DEFAULT_NONE,
+    ) -> IO:
+        """
+        Download this file. paramref:`out` needs to be supplied with a
+        :obj:`io.BufferedWriter`, the file contents will be saved to that object using the
+        :obj:`out.write<io.BufferedWriter.write>` method.
+
+        .. versionchanged:: 20.0
+
+            * :paramref:`custom_path` parameter now also accepts :class:`pathlib.Path` as argument.
+            * Returns :class:`pathlib.Path` object in cases where previously a :obj:`str` was
+              returned.
+            * Splitting :paramref:`custom_path` and :paramref:`out` in two functions.
+
+
+        Args:
+            out (:obj:`io.BufferedWriter`): A file-like object. Must be opened for writing in
```

Comment:
Re-Checking the python docs, maybe [BufferedIOBase](https://docs.python.org/3/library/io.html?highlight=bufferedwriter#io.BufferedIOBase) actually better represents what we mean here? Also the type hint should maybe be changed from `IO` to [`BytesIO`](https://docs.python.org/3/library/typing.html?highlight=typing%20io#typing.BinaryIO) ? 🤔

Reply:
They say a typical implemention should not inherit, but wrap though..?

Reply:
> A typical [BufferedIOBase](https://docs.python.org/3/library/io.html?highlight=bufferedwriter#io.BufferedIOBase) implementation should not inherit _from a [RawIOBase](https://docs.python.org/3/library/io.html?highlight=bufferedwriter#io.RawIOBase) implementation_, but wrap one, like [BufferedWriter](https://docs.python.org/3/library/io.html?highlight=bufferedwriter#io.BufferedWriter) and [BufferedReader](https://docs.python.org/3/library/io.html?highlight=bufferedwriter#io.BufferedReader) do.

and in the docs of `BufferedWriter` it says 

> It inherits [BufferedIOBase](https://docs.python.org/3/library/io.html?highlight=bufferedwriter#io.BufferedIOBase).

🤔

Reply:
Well, I don't know either xD

Reply:
I tried both ``issinstance`` and ``issubclass``, both are true, so we are good I guess.